crates/sui-core/src/transaction_driver/message_types.rs

```diff
@@ -49,8 +103,11 @@ pub struct QuorumTransactionResponse {
 }
 
 pub(crate) struct WaitForEffectsRequest {
-    pub consensus_position: ConsensusPosition,
     pub transaction_digest: TransactionDigest,
+    /// If provided, wait for the consensus position to execute and wait for fastpath outputs of the transaction,
+    /// in addition to waiting for finalized effects.
+    /// If not provided, only wait for finalized effects.
```

Comment:
Probably worth documenting that client should only skip the position if they have a reason to believe that this transaction has been finalized, since otherwise the quorum may be for different positions and hence invalid